cmd/opampsupervisor/specification/README.md

```diff
@@ -220,19 +225,69 @@ telemetry:
 
 ```
 
-**Note:**
+#### Notes on `agent::config_files`, `agent::args`, and `agent::env`
+
+Please be aware that when using the `agent::config_files` parameter,
+the configuration files specified are applied in the order they are specified.
+In other words, configuration files are merged from the top of the list to the bottom.
+Configuration added by files at the top of the list may be overwritten by the later ones.
+
+The indicated configuration files are merged in memory and the resulting configuration
+is written to `<storage::directory>/effective.yaml`.
 
-Please be aware that when using the `.agent.config_files` parameter,
-the configuration files specified are applied after the configuration from the OpAMP server.
-After the configuration files, arguments present in `.agent.args` are passed to the executable binary.
-The environmanet variables specified in `.agent.env` are set in the collector process environment.
+There are a few "magic" configuration files that can be used to completely
+customize the priority order of the Collector configuration. Below are the available
+values and what they represent:
 
-The following configuration:
+- `$OPAMP_EXTENSION_CONFIG`: configuration for the OpAMP extension to connect to the Supervisor.
+- `$OWN_METRICS_CONFIG`: configuration for the Collector to report its own metrics.
+- `$BUILTIN_CONFIG`: built-in base configuration.
+- `$REMOTE_CONFIG`: remote configuration received by the Supervisor.
+
+If **none** of these are explicitly specified, they are automatically prepended to the
```

Comment:
This would be a breaking change for the current behavior, but I would vote for having these **appended** to the list if they're unspecified:

1. The OpAMP extension configuration added by the Supervisor is essential for the Supervisor to work and disabling it should be something users have to explicitly opt into by putting the extension config before their custom config.
2. Ultimately, remote configuration is the "most specific" configuration, since it's been deliberately sent to a Collector by an operator. I think we should always respect this configuration unless the Collector has been specifically set up to override values that may be present in the remote configuration.